src/core/datatypes/trees/Tree.cpp

```diff
@@ -1172,21 +1172,8 @@ void Tree::initFromString(const std::string &s)
 {
     NewickConverter converter;
     Tree* bl_tree = converter.convertFromNewick( s );
-    if ( bl_tree->isUltrametric() == true )
```

Comment:
I'm wondering if this breaks some stuff. If I remember correctly, trees are always read in first using branch lengths. Then, only if they are assumed to be ultrametric, then we convert the branch lengths into ages. Maybe this happens somewhere  else now?

Reply:
I think when `initFromString` is called (presumably either when reinitializing from checkpoint or doing posterior predictive simulation), different from the `readTrees` or `readTreeTrace` functions, no `treetype` like argument would be provided (nor accessible from `initFromString`), so it wouldn't know whether the tree should be a timetree or not. However, as Sebastian pointed out, calling the `TreeUtilities::convertTree` function would be necessary if the tree needs to be a timetree, given that the return of `convertFromNewick` is a tree whose node ages have not been set.

In that case, perhaps the desired tree type could be determined by checking whether `*this` is a branch-length tree or a timetree (i.e., by checking the age of any given node in the tree, say the root node, is `nan` or not)? This should work if it is safe to assume that `initFromString` is called after the model DAG has been initialized (and thus `*this` should have been initialized to the desired tree type by the corresponding tree simulation function [if it's a stochastic node] or by assignment [if it's a constant node]).

Reply:
Thanks for the new commits @bredelings!

Related to another comment by Sebastian, as we now removed `root->setUseAges(true, true);` in the `convertFromNewick` function, perhaps we might want to add that to somewhere in the TreeUtilities::convertTree function? Currently the `bl_tree` will be created first by `convertFromNewick` and then converted to `time_tree` by `TreeUtilities::convertTree` if necessary, so the `use_ages` attribute of each TopologyNode will still be false in the finally converted timetree. Looks like this is probably innocuous given that we are not checking `use_ages` otherwhere, but perhaps it might be safer to make it consistent.

In addition, I just noticed that there is [one place](https://github.com/revbayes/revbayes/blob/development/src/core/analysis/mcmc/output/TreeSummary.cpp#L1318) we are calling `initFromString` where the tree has not been initialized so the tree type is not known. It looks like there we should know what the desired tree type is as the goal there is to check whether the tree converted from the string is contained in a certain credible interval of trees which is already read in as a `TraceTree` object. Perhaps we could check the type of first tree in the `TraceTree` to determine whether we would like the string to be converted to a timetree or not? Would be great to hear your thought on this @hoehna .